pkg/remote/google/repository/asset.go

```diff
@@ -75,101 +74,111 @@ func NewAssetRepository(client *asset.Client, config config.GCPTerraformConfig,
-	defer s.cache.Unlock(cacheKey)
-	if cachedResults != nil {
-		results = cachedResults.([]*assetpb.Asset)
-	}
+	filteredResults := []*assetpb.Asset{}
 
-	if results == nil {
-		it := s.client.ListAssets(context.Background(), req)
-		for {
-			resource, err := it.Next()
-			if err == iterator.Done {
-				break
-			}
-			if err != nil {
-				return nil, err
+	for _, scope := range s.config.Scope {
+		cacheKey := "listAllResources" + scope
+		cachedResults := s.cache.GetAndLock(cacheKey)
+		defer s.cache.Unlock(cacheKey)
+
+		req := &assetpb.ListAssetsRequest{
+			Parent:      scope,
+			ContentType: assetpb.ContentType_RESOURCE,
+			AssetTypes: []string{
+				cloudFunctionsFunction,
+				bigtableInstanceAssetType,
+				bigtableTableAssetType,
+				sqlDatabaseInstanceAssetType,
+				computeGlobalAddressAssetType,
+				nodeGroupAssetType,
+			},
+		}
+
+		var results []*assetpb.Asset
+
+		if cachedResults != nil {
+			results = cachedResults.([]*assetpb.Asset)
+		}
+
+		if results == nil {
+			it := s.client.ListAssets(context.Background(), req)
+			for {
+				resource, err := it.Next()
+				if err == iterator.Done {
+					break
+				}
+				if err != nil {
+					return nil, err
+				}
+				results = append(results, resource)
 			}
-			results = append(results, resource)
+			s.cache.Put(cacheKey, results)
 		}
-		s.cache.Put(cacheKey, results)
-	}
 
-	filteredResults := []*assetpb.Asset{}
-	for _, result := range results {
-		if result.AssetType == ty {
-			filteredResults = append(filteredResults, result)
+		for _, result := range results {
+			if result.AssetType == ty {
+				filteredResults = append(filteredResults, result)
+			}
 		}
 	}
-
 	return filteredResults, nil
 }
 
 func (s assetRepository) searchAllResources(ty string) ([]*assetpb.ResourceSearchResult, error) {
-	req := &assetpb.SearchAllResourcesRequest{
-		Scope: fmt.Sprintf("projects/%s", s.config.Project),
-		AssetTypes: []string{
-			storageBucketAssetType,
-			computeFirewallAssetType,
-			computeRouterAssetType,
-			computeInstanceAssetType,
-			computeNetworkAssetType,
-			computeSubnetworkAssetType,
-			dnsManagedZoneAssetType,
-			computeInstanceGroupAssetType,
-			bigqueryDatasetAssetType,
-			bigqueryTableAssetType,
-			computeAddressAssetType,
-			computeDiskAssetType,
-			computeImageAssetType,
-			healthCheckAssetType,
-			cloudRunServiceAssetType,
-		},
-	}
-	var results []*assetpb.ResourceSearchResult
 
-	cacheKey := "SearchAllResources"
-	cachedResults := s.cache.GetAndLock(cacheKey)
-	defer s.cache.Unlock(cacheKey)
-	if cachedResults != nil {
-		results = cachedResults.([]*assetpb.ResourceSearchResult)
-	}
+	filteredResults := []*assetpb.ResourceSearchResult{}
 
-	if results == nil {
-		it := s.client.SearchAllResources(context.Background(), req)
-		for {
-			resource, err := it.Next()
-			if err == iterator.Done {
-				break
-			}
-			if err != nil {
-				return nil, err
+	for _, scope := range s.config.Scope {
+		cacheKey := "SearchAllResources" + scope
```

Comment:
Could you please use the same pattern as for other repositories ?

```go
cacheKey := fmt.Sprintf("SearchAllResources_%s", scope)
```

Reply:
Done